src/worker/lambda/packages/packagePuller.go

```diff
@@ -21,6 +24,48 @@ import (
 // PackagePuller is the interface for installing pip packages locally.
 // The manager installs to the worker host from an optional pip
 // mirror.
+
+type SizeOfPackages struct {
+	TotalSize int            // total size of all packages
+	Packages  map[string]int // size of individual packages
+}
+
+// Initialize the global variable to track package sizes
+var sizeTracker = SizeOfPackages{
+	TotalSize: 0,
+	Packages:  make(map[string]int),
+}
+
+func PackageSizeIncreaser(pkg string, size int) {
+	// Normalize the package name
+	pkg = NormalizePkg(pkg)
+
+	// Lock the sizeTracker here if using in a concurrent context
+
+	// Update the size for the individual package and the total size
+	sizeTracker.Packages[pkg] += size
+	sizeTracker.TotalSize += size
+
+	// Unlock the sizeTracker here if using in a concurrent context
+}
+
+func getPackageSize(packageName string) (string, error) {
+	cmd := exec.Command("apt-cache", "show", packageName)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
```

Comment:
do we need to launch a separate program to check size?  That's been shown as an excessive overhead in other cases...